src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,261 @@
+        private Sampling _mode;
+        public const byte DefaultI2cAddress = 0x77;
+
+        public Bmp180(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;            
+            _calibrationData = new CalibrationData();            
+            //Read the coefficients table
+            _calibrationData.ReadFromDevice(this);
+            SetSampling(Sampling.Standard);
+        }
+        
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="mode">Sampling Mode</param>
+        public void SetSampling(Sampling mode)
+        {
+            _mode = mode;
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public Temperature ReadTemperature()
+        {
+            //Gets the compensated temperature in degrees celsius
+            short UT = ReadRawTemperature();
+
+            // Calculations below are taken straight from section 3.5 of the datasheet.
+            int X1 = (UT - _calibrationData.AC6) * _calibrationData.AC5 / (1 << 15);
+            int X2 = _calibrationData.MC * (1 << 11) / (X1 + _calibrationData.MD);
+            int B5 = (X1 + X2);
+            int T = (B5 + 8) / (1 << 4);
+            
+            return Temperature.FromCelsius((double)T/10);            
+        }
+
+        /// <summary>
+        ///  Reads the pressure from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Atmospheric pressure in Pa
+        /// </returns>
+        public int ReadPressure()
+        {
+            //Gets the compensated pressure in Pascals
+            short UT = ReadRawTemperature();
+            int UP = ReadRawPressure();
+
+            // Calculations below are taken straight from section 3.5 of the datasheet.
+            // Calculate true temperature coefficient B5.
+            int X1 = (UT - (short)_calibrationData.AC6) * (short)_calibrationData.AC5 / (1 << 15);
+            int X2 = (short)_calibrationData.MC * (1 << 11) / (X1 + (short)_calibrationData.MD);
+            int B5 = (X1 + X2);
+
+            // Pressure Calculations
+            int B6 = B5 - 4000;
+            X1 = ((short)_calibrationData.B2 * (B6 * B6 / (1 << 12))) / (1 << 11);
+            X2 = (short)_calibrationData.AC2 * B6 / (1 << 11);
+            int X3 = X1 + X2;
+            int B3 = ((((short)_calibrationData.AC1 * 4 + X3) << (short)Sampling.Standard) + 2) / 4;
+            X1 = (short)_calibrationData.AC3 * B6 / (1 << 13);
+            X2 = ((short)_calibrationData.B1 * (B6 * B6 / (1 << 12))) / (1 << 16);
+            X3 = ((X1 + X2) + 2) / (1 << 2);
+            int B4 = _calibrationData.AC4 * (X3 + 32768) / (1 << 15);
+            int B7 = (UP - B3) * (50000 >> (short)Sampling.Standard);
+
+            int p = 0;
+            if((uint)B7 < 0x80000000)

```

Comment:
not sure what's going on here - those below feel mathematically the same - is this some kind of handling for negative numbers? perhaps B7 should be already uint?